groups/bsl/bslstl/bslstl_error.h

```diff
@@ -710,7 +710,7 @@ struct hash<bsl::error_condition> : BloombergLP::bslh::Hash<>
 
 namespace std {
 
-#ifndef BSLS_PLATFORM_OS_DARWIN
+#if !defined(BSLS_PLATFORM_OS_DARWIN) || defined (BSLS_PLATFORM_CMP_GNU)
```

Comment:
"Compiling with gcc" seems to be a very big hammer - how did we not discover this issue earlier?
How does this code work without the template declaration? Are we getting it via a standard header, so we do not need to declare unless building with a  C++03 library? Perhaps that would be easier to check?

Reply:
It is *not* compiling with GCC.  It is compiling with gcc *on Apple*.